netbox/resource_netbox_site.go

```diff
@@ -130,8 +147,14 @@ func resourceNetboxSiteRead(d *schema.ResourceData, m interface{}) error {
 	d.Set("status", res.GetPayload().Status.Value)
 	d.Set("description", res.GetPayload().Description)
 	d.Set("facility", res.GetPayload().Facility)
-	d.Set("region_id", res.GetPayload().Region)
+	d.Set("region_id", res.GetPayload().Region.ID)
```

Comment:
You have to check for a nil region here as well.

~or you revert the region change~